app/javascript/components/DownloadButton.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faDownload } from '@fortawesome/free-solid-svg-icons';
+import Modal from 'react-bootstrap/lib/Modal';
+
+// TODO: Remove this once API endpoint is integrated
+import {authCodeResponseMock} from './FacetsMockData';
+
+function fetchDownloadConfig() {
+  const searchQuery = '&file_types=metadata,expression&accessions=SCP1,SCP2';
+  const authCode = authCodeResponseMock.auth_code; // Auth code is a one-time authorization token (OTAC)
+
+  const timeInterval = authCodeResponseMock.time_interval;
+
+  // Gets a curl configuration ("cfg.txt") containing signed
+  // URLs and output names for all files in the download object.
+  const url = (
+    window.origin +
+    '/api/v1/bulk_download?auth_code=' +
+    authCode +
+    searchQuery
+  );
+  const flag = (window.location.host === 'localhost') ? 'k' : ''; // "-k" === "--insecure"
+  
+  // This is what the user will run in their terminal to download the data.
+  const downloadCommand = (
```

Comment:
I had a conversation with Vicky yesterday about this - we may want to append `; rm cfg.txt` to the download command that the user runs.  Otherwise, they can _technically_ re-run the `curl -K cfg.txt` multiple times over the next day and download the data again as the URLs are valid for 24 hours.

Not a blocking concern, but possibly something we would like to think about.

Reply:
Good point, done in 619e805.  I manually verified it with single-study Bulk Download.